python/dgl/convert.py

```diff
@@ -0,0 +1,694 @@
+    data : graph data
+        Data to initialize graph structure. Supported data formats are
+        (1) list of edge pairs (e.g. [(0, 2), (3, 1), ...])
+        (2) pair of vertex IDs representing end nodes (e.g. ([0, 3, ...],  [2, 1, ...]))
+        (3) scipy sparse matrix
+        (4) networkx graph
+    utype : str, optional
+        Source node type name. (Default: _U)
+    etype : str, optional
+        Edge type name. (Default: _E)
+    vtype : str, optional
+        Destination node type name. (Default: _V)
+    card : pair of int, optional
+        Cardinality (number of nodes in the source and destination group). If None,
+        infer from input data.  (Default: None)
+    kwargs : key-word arguments, optional
+        Other key word arguments.
+
+    Returns
+    -------
+    DGLHeteroGraph
+    """
+    if utype == vtype:
+        raise DGLError('utype should not be equal to vtype. Use ``dgl.graph`` instead.')
+    if card is not None:
+        urange, vrange = card
+    else:
+        urange, vrange = None, None
+    if isinstance(data, tuple):
+        u, v = data
+        return create_from_edges(u, v, utype, etype, vtype, urange, vrange)
+    elif isinstance(data, list):
+        return create_from_edge_list(data, utype, etype, vtype, urange, vrange)
+    elif isinstance(data, sp.sparse.spmatrix):
+        return create_from_scipy(data, utype, etype, vtype)
+    elif isinstance(data, nx.Graph):
+        return create_from_networkx_bipartite(data, utype, etype, vtype, **kwargs)
+    else:
+        raise DGLError('Unsupported graph data type:', type(data))
+
+def hetero_from_relations(rel_graphs):
+    """Create a heterograph from per-relation graphs.
+
+    TODO(minjie): this API can be generalized as a union operation of
+    the input graphs
+
+    TODO(minjie): handle node/edge data
+
+    Parameters
+    ----------
+    rel_graphs : list of DGLHeteroGraph
+        Graph for each relation.
+
+    Returns
+    -------
+    DGLHeteroGraph
+        A heterograph.
+    """
+    # infer meta graph
+    ntype_dict = {}  # ntype -> ntid
+    meta_edges = []
+    ntypes = []
+    etypes = []
+    for rgrh in rel_graphs:
+        assert len(rgrh.etypes) == 1
+        stype, etype, dtype = rgrh.canonical_etypes[0]
+        if not stype in ntype_dict:
+            ntype_dict[stype] = len(ntypes)
+            ntypes.append(stype)
+        stid = ntype_dict[stype]
+        if not dtype in ntype_dict:
+            ntype_dict[dtype] = len(ntypes)
+            ntypes.append(dtype)
+        dtid = ntype_dict[dtype]
+        meta_edges.append((stid, dtid))
+        etypes.append(etype)
+    metagraph = graph_index.from_edge_list(meta_edges, True, True)
+    # create graph index
+    hgidx = heterograph_index.create_heterograph_from_relations(
+        metagraph, [rgrh._graph for rgrh in rel_graphs])
+    return DGLHeteroGraph(hgidx, ntypes, etypes)
+
+def hetero_from_homo(G, ntypes, etypes, ntype_field='type', etype_field='type'):
```

Comment:
We should be consistent with `hetero_from_homo` and set the default values of `ntype_field` and `etype_field` to be `NTYPE` and `ETYPE`.

Reply:
Done